test/lsp_test_runner.cc

```diff
@@ -700,58 +716,142 @@ TEST_CASE("LSPTest") {
             auto assertions = RangeAssertion::parseAssertions(updatesAndContents);
             auto assertFastPath = FastPathAssertion::get(assertions);
             auto assertSlowPath = BooleanPropertyAssertion::getValue("assert-slow-path", assertions);
-            auto responses = getLSPResponsesFor(*lspWrapper, move(lspUpdates));
-            bool foundTypecheckRunInfo = false;
-
-            for (auto &r : responses) {
-                if (r->isNotification()) {
-                    if (r->method() == LSPMethod::SorbetTypecheckRunInfo) {
-                        auto &params = get<unique_ptr<SorbetTypecheckRunInfo>>(r->asNotification().params);
-                        // Ignore started messages. Note that cancelation messages cannot occur in test_corpus since
-                        // test_corpus only runs LSP in single-threaded mode.
-                        if (params->status == SorbetTypecheckRunStatus::Ended) {
-                            foundTypecheckRunInfo = true;
-                            if (assertSlowPath.value_or(false)) {
-                                INFO(errorPrefix << "Expected Sorbet to take slow path, but it took the fast path.");
+
+            // TODO(aprocter): There's probably more code duplication than necessary between the 'if' and the 'else'
+            // here.
+            if (haveStaleUpdates) {
+                // Wait for a fence, to clear out any pending slow-path operations.
+                (void)getLSPResponsesFor(*lspWrapper, std::vector<std::unique_ptr<LSPMessage>>{});
```

Comment:
We don't need to loop here to account for potential delays in receiving the typechecking done notification?

Reply:
This should clear out the typechecking-done notification, since it will be sent as part of the slowpath run triggered in the edit task. (It seemed to fix the problem we had been seeing, anyway :))